### PR TITLE
Add finalizer to DeflateStream

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/mono/mono/issues/15825

### CODE REVIEW

File: mcs/class/System/System.IO.Compression/DeflateStream.cs [hunk 1]
```diff
@@ -104,8 +104,13 @@ internal DeflateStream (Stream stream, CompressionLevel compressionLevel, bool l
 		{
 		}		
 
+		~DeflateStream () => Dispose (false);
+
 		protected override void Dispose (bool disposing)
 		{
+			if (disposing)
+				GC.SuppressFinalize (this);
```
Comment (CoffeeFlux) [inline mcs/class/System/System.IO.Compression/DeflateStream.cs]: Wouldn't this go in the next conditional? The logic around the disposing is hard for me to follow, so maybe I'm misunderstanding.

**Comment by CoffeeFlux** [comment]
Also, this seems like something worth backporting since people it's been reported multiple times now, though I don't know how that works with the API diff since it's technically a change.

File: mcs/class/System/System.IO.Compression/DeflateStream.cs (hunk 1 above)
Comment (EgorBo) [reply mcs/class/System/System.IO.Compression/DeflateStream.cs]: @CoffeeFlux it's called "dispose pattern", see https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implementing-the-dispose-pattern-for-a-base-class

**Comment by EgorBo** [comment]
@monojenkins build failed

**Comment by steveisok** [comment]
@monojenkins build failed

**Comment by marek-safar** [comment]
@monojenkins backport 2019-06

**Comment by marek-safar** [comment]
@monojenkins backport 2019-02

File: mcs/class/System/System.IO.Compression/DeflateStream.cs [hunk 2]
```diff
@@ -104,8 +104,13 @@ internal DeflateStream (Stream stream, CompressionLevel compressionLevel, bool l
 		{
 		}		
 
+		~DeflateStream () => Dispose (false);
```
Comment (marek-safar) [inline mcs/class/System/System.IO.Compression/DeflateStream.cs]: This is not .NET Framework compatible fix but I am fine with it for now

**Comment by marek-safar** [comment]
@monojenkins commit apidiff

**Pull request merged**